cmd/bootstrapper/bootstrapper.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/timesync"
+)
+
+var (
+	bitcoinUrl     string
+	nodeEndpoint   string
+	genesisTime    string
+	layerDuration  time.Duration
+	layersPerEpoch uint32
+	epochOffset    uint32
+	serveUpdate    bool
+	port           int
+	dataDir        string
+	logLevel       string
+	logjson        bool
+)
+
+func init() {
+	// external resources
+	cmd.PersistentFlags().StringVar(&bitcoinUrl, "bitcoin-url",
+		"https://api.blockcypher.com/v1/btc/main", "specify URL to get bitcoin block hash")
+	cmd.PersistentFlags().StringVar(&nodeEndpoint, "node-endpoint",
+		"localhost:9092", "specify grpc endpoint for a spacemesh node")
+
+	// network parameters
+	cmd.PersistentFlags().StringVar(&genesisTime, "genesis-time",
+		"", "Time of the genesis layer in 2019-13-02T17:02:00+00:00 format")
+	cmd.PersistentFlags().DurationVar(&layerDuration, "layer-duration",
+		5*time.Minute, "duration per layer")
+	cmd.PersistentFlags().Uint32Var(&layersPerEpoch, "layers-per-epoch",
+		4032, "number of layers per epoch")
```

Comment:
it was definitely better to get all this data from the node

Reply:
done